### DISCUSSION TITLE
BDD Mockito vs Mockito

### DISCUSSION THREAD

**Discussion by nayonsoso** [solid-connection/solid-connect-server#134] [논의]
둘 중 어느 것을 사용하는게 좋을까요? 컨벤션을 정해봅시다!

**Comment by Gyuhyeok99** [comment]
### BDD Mockito vs Mockito
지난주 회의를 통해 느낀 점은 테스트 코드 작성에서 중요한 요소 중 하나가 가독성이라는 것이었습니다. 이를 바탕으로, 두 가지 접근 방식(BDD Mockito와 일반 Mockito)을 비교해보았습니다.

1️⃣ BDD Mockito: 구조적이고 협업에 적합한 방식
BDD Mockito는 given, when, then 키워드를 사용하여 **행위 중심(Behavior-Driven)**의 테스트 코드를 작성

장점: 테스트 코드를 읽는 사람이 시나리오에 맞게 테스트 코드가 읽힐 수 있도록 흐름을 자연스럽게 이해할 수 있음
단점: 코드가 길어질 수 있고 새로 학습해야하기에 추가 학습 비용 발생

2️⃣  일반 Mockito: 익숙하고 간결한 방식
일반 Mockito는 when과 verify를 사용하여 단위 테스트 중심의 테스트 코드를 작성
장점: 단순한 테스트에 적합
단점: 코드가 단순하더라도 테스트 의도를 명확히 드러내지 못할 수 있음

### 최종 정리
BDD Mockito는 테스트가 시나리오 기반으로 읽히도록 작성할 수 있다는 점에서 장점이 있습니다.
테스트 코드를 협업의 도구로 활용하고자 할 때, BDD Mockito의 구조적 접근 방식은 가독성과 의도 전달 면에서 더 큰 이점을 제공합니다.

따라서, 비즈니스 로직 중심의 테스트 환경에서 단위 테스트가 필요하다면 BDD Mockito를 선택하는 방식으로 컨벤션을 정하면 좋을 것 같습니다. 😊

**Comment by nayonsoso** [comment]
저도 BDD Mockito 를 사용하자는 쪽입니다.
given - when - then 구조로 테스트를 작성하고 있기 때문에 BDD Mockito 가 더 적합하다고 생각합니다!
